src/mono/wasm/runtime/startup.ts

```diff
@@ -56,6 +56,42 @@ export function mono_wasm_set_runtime_options(options: string[]): void {
     cwraps.mono_wasm_parse_runtime_options(options.length, argv);
 }
 
+async function _fetch_asset(url: string): Promise<Response> {
+    try {
+        if (typeof (fetch) === "function") {
+            return fetch(url, { credentials: "same-origin" });
+        }
+        else if (ENVIRONMENT_IS_NODE) {
+            // eslint-disable-next-line @typescript-eslint/no-var-requires
+            const fs = require("fs");
```

Comment:
This will do `require` for each call of the `_fetch_asset`. I admit the previous implementation had the same issue.
Also I strongly prefer the async variant. This is "stop the world and wait for disk" for node engine.